File: day01/main.py
#!/usr/bin/python3

def solve_captcha(puzzle_input, step):
    sum = 0
    for index, value in enumerate(puzzle_input):
        next_index = (index + step) % len(puzzle_input)
        if value == puzzle_input[next_index]:
            sum += value
    return sum

if __name__ == "__main__":
    puzzle_input = open("input.txt").read()
    puzzle_input = list(map(int, puzzle_input))

    # Phase 1
    print(solve_captcha(puzzle_input, 1))

    # Phase 2
    print(solve_captcha(puzzle_input, len(puzzle_input) // 2))

File: day01/main.go
package main

import (
	"fmt"
	"io/ioutil"
	"strings"
	"strconv"
)

func solveCaptcha(puzzleInput []int, step int) int {
	sum := 0
	for index, value := range puzzleInput {
		nextIndex := (index + step) % len(puzzleInput)
		if value == puzzleInput[nextIndex] {
			sum += value
		}
	}
	return sum
}

func main() {
	inputFile, err := ioutil.ReadFile("input.txt")
	if err != nil {
		panic(err)
	}
	stringInput := strings.Split(string(inputFile), "")
	puzzleInput := make([]int, len(stringInput))
	for index, value := range stringInput {
		i, err := strconv.Atoi(value)
		if err != nil {
			panic(err)
		}
		puzzleInput[index] = i
	}

	// Phase 1
	fmt.Println(solveCaptcha(puzzleInput,1))

	// Phase 2
	fmt.Println(solveCaptcha(puzzleInput, len(puzzleInput) / 2))
}